### PR TITLE
docs: Update Kubernetes microsite documentation

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

Noticed the Kubernetes page in the microsite had a duplicated header. Went in to fix that, and just took a quick stab to improving the page a bit.

Note: refactored the reference to the cluster `serviceAccountToken` to use `$env`, so as not to recommend exposing those creds directly in the config. I've not actually done this, but happy to be corrected if that's not functional, but figure the example should be what we would expect a user should do?

| Before | After |
| --- | --- |
| image | image |

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by adamdmharvey** [comment]
cc @AlleyWei as I noticed you have a pending blog post, this may help align

**Review by mclarke47 — APPROVED** [review]

**Comment by mclarke47** [comment]
LGTM! Thanks!

File: docs/features/software-catalog/kubernetes-in-backstage.md [hunk 1]
```diff
@@ -78,7 +74,7 @@ cluster. Valid values are:
 The service account token to be used when using the `serviceAccount` auth
 provider.
 
-## RBAC
+## Roll Based Access Control
```
Comment (freben) [inline docs/features/software-catalog/kubernetes-in-backstage.md:77]:
```suggestion
## Role Based Access Control
```

**Review by freben — APPROVED** [review]
One minor tweak needed, then good to go

File: docs/features/software-catalog/kubernetes-in-backstage.md (hunk 1 above)
Comment (adamdmharvey) [reply docs/features/software-catalog/kubernetes-in-backstage.md]: Clearly, I was on a roll 🎡 and missed this... 🙄 🤣

**Pull request merged**